api/src/Entity/ContentNode.php

```diff
@@ -70,6 +71,7 @@ abstract class ContentNode extends BaseEntity implements BelongsToContentNodeTre
     #[Assert\NotNull(groups: ['create'])] // Root nodes have parent:null, but manually creating root nodes is not allowed
     #[AssertNoRootChange(groups: ['update'])]
     #[AssertNoLoop(groups: ['update'])]
+    #[AssertAttachedToRoot(groups: ['update'])]
```

Comment:
why isn't this validation only on the DefaultLayout,
like the additional validations on the data attribute in the ContentNode child classes?

Reply:
Because I thought that attributes don't get applied to inherited fields. So I would need to add all these validators from the ContentNode parent fields and I thought this way it is easier to maintain.

Reply:
ah ok. WIth data the validations are very different.
Here you only want to add one additional validation for one child class.
There would be solutions, but we can do that later.